app/delegate-server/Lib/Transaction.purs

```diff
@@ -0,0 +1,55 @@
+module DelegateServer.Lib.Transaction
+  ( reSignTransaction
+  , setExUnitsToMax
+  , setTxValid
+  ) where
+
+import Prelude
+
+import Contract.Monad (Contract)
+import Contract.ProtocolParameters (getProtocolParameters)
+import Contract.Transaction
+  ( BalancedSignedTransaction
+  , Language(PlutusV2)
+  , Transaction
+  , _isValid
+  , _plutusData
+  , _vkeys
+  , _witnessSet
+  , signTransaction
+  )
+import Ctl.Internal.Cardano.Types.Transaction (_redeemers)
+import Ctl.Internal.Transaction (setScriptDataHash)
+import Data.Lens ((^.), (%~), (.~))
+import Data.Lens.Common (simple)
+import Data.Lens.Iso.Newtype (_Newtype)
+import Data.Map (filterKeys) as Map
+import Data.Maybe (Maybe(Nothing), fromMaybe)
+import Data.Newtype (modify, unwrap, wrap)
+import Effect.Class (liftEffect)
+
+reSignTransaction :: BalancedSignedTransaction -> Contract BalancedSignedTransaction
+reSignTransaction tx =
+  signTransaction
+    (tx # simple _Newtype <<< _witnessSet <<< _vkeys .~ Nothing)
+
+setTxValid :: Transaction -> Transaction
+setTxValid = _isValid .~ true
+
+setExUnitsToMax :: Transaction -> Contract Transaction
```

Comment:
Why do we need that?

Reply:
This function sets the transaction execution units to the maximum value defined in the protocol parameters. It is only used for PlaceBid L2 transactions. These transactions involve spending L2 utxos that are not available on L1, which makes tx evaluation fail, but since execution units are free on Hydra, we simply skip the evaluation step and set the execution units manually.